src/nvim/window.c

```diff
@@ -1687,15 +1680,15 @@ int make_windows(int count, bool vertical)
   int todo;
 
   if (vertical) {
-    // Each window needs at least 'winminwidth' lines and a separator
-    // column.
+    // Each window needs at least 'winminwidth' lines and a separator column.
     maxcount = (curwin->w_width + curwin->w_vsep_width
                 - (p_wiw - p_wmw)) / (p_wmw + 1);
   } else {
-    // Each window needs at least 'winminheight' lines
-    // If statusline isn't global, each window also needs a statusline
+    // Each window needs at least 'winminheight' lines.
+    // If statusline isn't global, each window also needs a statusline.
+    // If 'winbar' is set, each window also needs a winbar.
     maxcount = (curwin->w_height + curwin->w_hsep_height + curwin->w_status_height
-                - (p_wh - p_wmh)) / (p_wmh + STATUS_HEIGHT);
+                - (p_wh - p_wmh)) / (p_wmh + STATUS_HEIGHT + winbar_height());
```

Comment:
shouldn't this use the possibly local value? `winbar_height()` could take a `win_T *` arg and check both global and local 'winbar' value (but check NULL and then only use global value).

Reply:
ok, so the interpretation of `p_wmh` will only be regards with the global value. I'd suggest renaming the function `global_winbar_height()` as not to give a false impression (and avoid ns clash with `winbar_height` variables in the same file)